src/App.tsx

```diff
@@ -15,48 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+interface SortFields {
+  sortField: SortType,
+  isReverse: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReverse } :SortFields,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.Default) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return goodA.localeCompare(goodB);
+
+        case SortType.Length:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+  const visibleGoods: string[] = getPreparedGoods(goodsFromServer,
+    { sortField, isReverse });
```

Comment:
Why are you putting the last two arguments into the object?

If arguments order bother you, then it's better to put everything in an object.

Reply:
> Why are you putting the last two arguments into the object?
> 
> If arguments order bother you, then it's better to put everything in an object.


Thanks for the advice. But I don’t completely agree with you, because I think that the array is always there, and the sorting parameters can change and expand, and it’s better to pass them as an object.